backend/src/main/java/org/cryptomator/hub/validation/AlmostSafeText.java

```diff
@@ -0,0 +1,29 @@
+package org.cryptomator.hub.validation;
+
+import javax.validation.Constraint;
+import javax.validation.Payload;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.Pattern;
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.ANNOTATION_TYPE;
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.TYPE_USE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+@Pattern(regexp = "(?Ui)[-_\\w\\s]+")
+@NotBlank
+@Target({METHOD, FIELD, ANNOTATION_TYPE, TYPE_USE})
+@Retention(RUNTIME)
+@Constraint(validatedBy = {})
+@Documented
+public @interface AlmostSafeText {
```

Comment:
This is a terrible name. Looking at the usage, I would suggest renaming this to something like "UserFacingText" and renaming SafeText to ValidID